.github/actions/e2e/setup-cluster/action.yaml

```diff
@@ -215,19 +215,23 @@ runs:
       # Adding taints after all necessary pods have scheduled to the manged node group nodes
       # amazon-cloudwatch-observability pods do no not tolerate CriticalAddonsOnly=true:NoSchedule and
       # amazon-cloudwatch-observability addons does not allow to add tolerations to the addon pods as part of the advanced configuration
-      kubectl taint nodes CriticalAddonsOnly=true:NoSchedule --all
+      # Overwrite existing taints to ensure we don't fail here on upgrade
+      kubectl taint nodes CriticalAddonsOnly=true:NoSchedule --all --overwrite
 
-      # We delete DaemonSets that we don't care about because it causes inconsistencies in scheduling due to
-      # dcgm-exporter and neuron-monitor selecting on specific instance types
-      # See https://github.com/kubernetes-sigs/karpenter/issues/715 for more detail
-      kubectl delete daemonsets -n amazon-cloudwatch dcgm-exporter neuron-monitor
+      # Only modify cloudwatch daemonsets if the addon was installed. This is not guaranteed in the upgrade test since only the control plane is upgraded.
+      if [[ "$(aws eks list-addons --cluster "${CLUSTER_NAME}" | yq '.addons | contains(["amazon-cloudwatch-observability"])')" == "true" ]]; then
```

Comment:
Doesn't this mean that we won't have the CWA addon in the upgrade test? Should we modify our code so that we still get the CWA installed even in the upgrade scenario? Maybe call specific calls on eksctl in the upgrade test to ensure that they get installed if they aren't already there.

Reply:
I think I agree that this is the better approach, I've updated the PR. Not sure it matters too much in the long run since eventually the source cluster will have the cloudwatch addon as well but this probably makes sense for us to ensure generally anyway.